### DISCUSSION TITLE
Grain growth actions

### DISCUSSION THREAD

**Discussion by xueyang94** [idaholab/moose#21478] [Q&A Modules: Phase field] [answered]
I am new to modeling grain growth. I saw that there are two actions available: the GrainGrowthAction and PolycrystalKernalAction. What are they major differences between the two? Also, I have a phase-field model with 5 phases. I want to add polycrystal structure to only one of the phases by Voronoi Tessellation. From what I understand, adding the microstructure only adds more phase parameters and interfacial energy. I know how to generate Voronoi Tessellation in the entire domain, but how to do it in only one phase? Thanks.

**Comment by GiudGiud** [comment]
@SudiptaBiswas @dschwen

**Comment by SudiptaBiswas** [comment, marked as answer]
`PolycrystalKernalAction` only adds the kernels needed for grain growth simulations, whereas `GrainGrowthAction` also adds the pieces needed to visualize the outputs. For your next question, you might need to modify the initial condition to do what you need. It would depend on how you are initializing different phases. For example, look into the `PolycrystalVoronoiVoidIC` regarding how to include a void phase.

**Reply by xueyang94** [reply]
@SudiptaBiswas Thank you for helping. What outputs can GrainGrowthAction visualize other than bnds? Will GrainGrowthAction completely replace PolycrystalKernalAction? 

For the next part, I tried using `SubdomainBoundingBoxGenerator` to get a block of mesh for assigning the grain structure because the phase has a rectangular geometry instead of a void. How is the interfacial width between two domains of mesh handled?
